lib/RoleCredentialResolver.ts

```diff
@@ -0,0 +1,164 @@
+      | undefined;
+    credential = await this.getVerifiableCredential(did, namespace);
+    if (!credential) {
+      credential = await this.getEIP191JWT(did, namespace);
+    }
+    return credential;
+  }
+
+  /**
+   * Fetches Verifiable Credential for the given did and role for a vc issuance hierarchy
+   *
+   * ```typescript
+   * const credentialResolver = new IpfsCredentialResolver(
+   *  provider,
+   *  registrySettings,
+   *  didStore );
+   * const credential = credentialResolver.getVerifiableCredential('did:ethr:1234', 'sampleRole');
+   * ```
+   *
+   * @param did subject DID for which the credential needs to be fetched
+   * @param namespace role for which the credential needs to be fetched
+   * @returns
+   */
+  async getVerifiableCredential(did: string, namespace: string) {
+    const credentials = await this.credentialsOf(did);
+    return credentials.find(
+      (claim) =>
+        claim.credentialSubject.role.namespace === namespace ||
+        utils.namehash(claim.credentialSubject.role.namespace) === namespace
+    );
+  }
+
+  /**
+   * Fetches RoleEIP191JWT for the given did and role for an OffChainClaim issuance hierarchy
+   *
+   * ```typescript
+   * const credentialResolver = new IpfsCredentialResolver(
+   *  provider,
+   *  registrySettings,
+   *  didStore );
+   * const credential = credentialResolver.getEIP191JWT('did:ethr:1234', 'sampleRole');
+   * ```
+   *
+   * @param did subject DID for which the credential to be fetched
+   * @param role role for which the credential need to be fetched
+   * @returns RoleEIP191JWT
+   */
+  async getEIP191JWT(
+    did: string,
+    namespace: string
+  ): Promise<RoleEIP191JWT | undefined> {
+    const eip191Jwts = await this.eip191JwtsOf(did);
+    return eip191Jwts.find(
+      (jwt) =>
+        jwt?.payload?.claimData.claimType === namespace ||
+        utils.namehash(jwt?.payload?.claimData.claimType) === namespace
+    );
+  }
+
+  /**
+   * Fetches all the Role eip191Jwts belonging to the subject DID
+   * @param did subject DID
+   * @returns RoleEIP191JWT list
+   */
+  async eip191JwtsOf(did: string): Promise<RoleEIP191JWT[]> {
+    if (this._cacheServerClient?.isAvailable) {
+      const roleEIP191Jwts =
+        await this._cacheServerClient.getRoleEIP191JwtsBySubject(did, {
+          isAccepted: true,
+        });
+      return roleEIP191Jwts
+        .filter(isEIP191Jwt)
+        .map(transformClaim)
+        .filter(filterOutMaliciousClaims);
+    }
+    return await this._ipfsCredentialResolver.eip191JwtsOf(did);
+  }
+
+  /**
+   * Fetches all the Verifiable Credential belonging to the subject DID
+   * @param did subject DID
+   * @returns VerifiableCredential<RoleCredentialSubject> list
+   */
+  async credentialsOf(
+    did: string
+  ): Promise<VerifiableCredential<RoleCredentialSubject>[]> {
+    if (this._cacheServerClient?.isAvailable) {
+      const vcs =
+        await this._cacheServerClient.getVerifiableCredentialBySubject(did, {
```

Comment:
Thinking about it a bit more, I think this might be a problem as the identity of running passport-did-auth will likely not be able to access all of the credentials of the issuer due to authorization protection on ssi-hub. On ssi-hub, only a valid issuer or the subject can read the subject's credentials, I believe.